### PR TITLE
Adds support for a new attribute, 'relType', for the @Relationship annotation.

### PR DESCRIPTION
**Pull request by emetsger**
Addresses issue #21.

### CODE REVIEW

File: src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java [hunk 1]
```diff
@@ -351,12 +360,14 @@ private void handleRelationships(JsonNode source, Object object, Map<String, Obj
 
 					// Use resolver if possible
 					if (resolveRelationship && resolver != null && relationship.has(LINKS)) {
-						JsonNode self = relationship.get(LINKS).get(SELF);
+						String relType = RELATIONSHIP_RELTYPE_MAP.get(object.getClass())
```
Comment (jasminb) [inline src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java]: You can use `relationshipField` to get the `Relationship` from the `FIELD_RELATIONSHIP_MAP`:

```
String relType = FIELD_RELATIONSHIP_MAP.get(relationshipField).getRelType();
```

File: src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java [hunk 2]
```diff
@@ -69,6 +71,13 @@ public ResourceConverter(ObjectMapper mapper, Class<?>... classes) {
 					RELATIONSHIP_TYPE_MAP.get(clazz).put(relationship.value(), targetType);
 					RELATIONSHIP_FIELD_MAP.get(clazz).put(relationship.value(), relationshipField);
 					FIELD_RELATIONSHIP_MAP.put(relationshipField, relationship);
+					if (relationship.resolve() && relationship.relType() == null) {
+						throw new IllegalArgumentException("@Relationship on " + clazz.getName() + "#" +
+								relationshipField.getName() + " with 'resolve = true' must have a relType attribute " +
+								"set." );
+					} else if (relationship.resolve()) {
+						RELATIONSHIP_RELTYPE_MAP.get(clazz).put(relationship.value(), relationship.relType());
```
Comment (jasminb) [inline src/main/java/com/github/jasminb/jsonapi/ResourceConverter.java]: See comment below, `RELATIONSHIP_RELTYPE_MAP` can be removed.

**Comment by emetsger** [comment]
Ah yes, fixed: 4195f76

**Comment by jasminb** [comment]
Thanks!

**Comment by emetsger** [comment]
Thank you!  This library is going to save us a lot of time :)

**Pull request merged**